### DISCUSSION TITLE
Possibility of full interaction with Telegram Web App

### DISCUSSION THREAD

**Discussion by CrazyTapok-bit** [westacks/telebot#85] [Ideas]
Today, users increasingly need to use [Telegram Login Widget](https://core.telegram.org/widgets/login) and [Telegram Web App](https://core.telegram.org/bots/webapps) in their projects. So I have a great suggestion to connect the new, lightweight and very powerful [tgWebValid](https://github.com/CrazyTapok-bit/tgWebValid) library to your library for validating such users. This will not only expand the capabilities of your library, but also attract a significant number of new users to its use. After all, this is a strong impetus for further improvement of open projects. I look forward to your feedback and thoughts on this idea!

**Comment by punyflash** [comment]
Thanks a lot, but webapp support and authentication already implemented in TeleBot

**Reply by CrazyTapok-bit** [reply]
Yes, I understand that. But we can make this process better by adding autocomplete and a few other useful things. Shall we talk?

**Reply by punyflash** [reply]
TeleBot is open for contributions in case you have some ideas. However, I'm not into relying on external packages except Guzzle.
